server/accounts.go

```diff
@@ -2846,20 +2844,26 @@ func (s *Server) updateAccountClaimsWithRefresh(a *Account, ac *jwt.AccountClaim
 	// Reset any notion of export revocations.
 	a.actsRevoked = nil
 
+	alteredScope := map[string]struct{}{}
+
 	// update account signing keys
 	a.signingKeys = nil
+	if len(ac.SigningKeys) > 0 {
+		a.signingKeys = make(map[string]jwt.Scope, len(ac.SigningKeys))
+	}
 	signersChanged := false
-	for k := range ac.SigningKeys {
-		a.signingKeys = append(a.signingKeys, k)
+	for k, scope := range ac.SigningKeys {
+		a.signingKeys[k] = scope
 	}
-	sort.Strings(a.signingKeys)
 	if len(a.signingKeys) != len(old.signingKeys) {
```

Comment:
this doesn't look right - the count of the keys may be the same but the keys could have changed?

Reply:
That's what the else is doing. This basic check is to say that if count is different, there is surely something that has changed. However, since now we keep track of the one that changed, we may need to do the "else" case anyway.

Reply:
not the same but when different we don't compare the scope. fixed.